presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcConnector.java

```diff
@@ -31,30 +39,63 @@
     private static final Logger log = Logger.get(JdbcConnector.class);
 
     private final LifeCycleManager lifeCycleManager;
-    private final JdbcMetadata jdbcMetadata;
+    private final JdbcMetadataFactory jdbcMetadataFactory;
     private final JdbcSplitManager jdbcSplitManager;
     private final JdbcRecordSetProvider jdbcRecordSetProvider;
     private final JdbcRecordSinkProvider jdbcRecordSinkProvider;
 
+    private final ConcurrentMap<ConnectorTransactionHandle, JdbcMetadata> transactions = new ConcurrentHashMap<>();
+
     @Inject
     public JdbcConnector(
             LifeCycleManager lifeCycleManager,
-            JdbcMetadata jdbcMetadata,
+            JdbcMetadataFactory jdbcMetadataFactory,
             JdbcSplitManager jdbcSplitManager,
             JdbcRecordSetProvider jdbcRecordSetProvider,
             JdbcRecordSinkProvider jdbcRecordSinkProvider)
     {
         this.lifeCycleManager = requireNonNull(lifeCycleManager, "lifeCycleManager is null");
-        this.jdbcMetadata = requireNonNull(jdbcMetadata, "jdbcMetadata is null");
+        this.jdbcMetadataFactory = requireNonNull(jdbcMetadataFactory, "jdbcMetadataFactory is null");
         this.jdbcSplitManager = requireNonNull(jdbcSplitManager, "jdbcSplitManager is null");
         this.jdbcRecordSetProvider = requireNonNull(jdbcRecordSetProvider, "jdbcRecordSetProvider is null");
         this.jdbcRecordSinkProvider = requireNonNull(jdbcRecordSinkProvider, "jdbcRecordSinkProvider is null");
     }
 
     @Override
-    public ConnectorMetadata getMetadata()
+    public boolean isSingleStatementWritesOnly()
+    {
+        return true;
+    }
+
+    @Override
+    public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel, boolean readOnly)
+    {
+        checkConnectorSupports(READ_COMMITTED, isolationLevel);
```

Comment:
This change is just to update to the API. Supporting anything else is a much more involved architectural change.